File: 136-single-number/136-single-number.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # Solution: O(n) time, O(1) additional space.
        #   XOR of duplicate numbers zero-out, so result after XORing all nums is the one with no duplicate.
        result = nums[0]
        for i in range(1, len(nums)):
            result ^= nums[i]
        return result

File: 136-single-number/136-single-number.java
class Solution {
    
    public int singleNumber(int[] nums) {
        
        // Solution: O(n) time, O(1) additional space.
        
        int result = nums[0];
        
        // XOR of duplicate numbers zero-out, so result after XORing all nums is the one with no duplicate.
        for (int i = 1; i < nums.length; i++) result ^= nums[i]; 
         
        return result;
        
    }
    
}